lib/jxl/dec_patch_dictionary.h

```diff
@@ -125,20 +127,23 @@ class PatchDictionary {
   std::vector<PatchReferencePosition> ref_positions_;
   std::vector<PatchBlending> blendings_;
 
-  // TODO(szabadka) Limit the size of these vectors or use an interval tree.
-  // Patch occurrences sorted by y.
-  std::vector<size_t> sorted_patches_;
-  // Index of the first patch for each y value.
-  std::vector<size_t> patch_starts_;
-
-  // Patch IDs in position [patch_starts_[y], patch_start_[y+1]) of
-  // sorted_patches_ are all the patches that intersect the horizontal line at
-  // y.
-  // The relative order of patches that affect the same pixels is the same -
-  // important when applying patches is noncommutative.
-
-  // Compute patches_by_y_ after updating positions_.
-  void ComputePatchCache();
+  // Interval tree on the y coordinates of the patches.
+  struct PatchTreeNode {
+    int left_child;
+    int right_child;
+    size_t y_center;
```

Comment:
What's the rationale with y_center being size_t, but left_child and right_child not?

Reply:
I use -1 for the case that there is no left/right child

Reply:
Then it should be int64_t/ssize_t, no? (or the size_t should be uint32_t / unsigned int)

Reply:
Ok, I made these ssize_t and also tree_idx in the lookup and the int64_ts in the reverse-loop for consistency.